tests/test_end_to_end.py

```diff
@@ -0,0 +1,57 @@
+"""An end to end example and test."""
+from typing import Callable
+
+from sklearn import datasets, svm
+from sklearn.model_selection import cross_val_score
+
+from kotsu import run
+from kotsu.registration import ModelRegistry, ValidationRegistry
+
+
+model_registry = ModelRegistry()
+
+model_registry.register(
+    id="SCV-v1",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 1, "random_state": 1},
+)
+
+model_registry.register(
+    id="SCV-v2",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 0.5, "random_state": 1},
+)
+
+validation_registry = ValidationRegistry()
+
+
+def factory_iris_cross_val(folds: int) -> Callable:
+    """Factory for iris cross validation."""
+
+    def iris_cross_val(model) -> dict:
+        """Iris classification cross validation."""
+        X, y = datasets.load_iris(return_X_y=True)
+        scores = cross_val_score(model, X, y, cv=folds)
```

Comment:
this is another example where it could be useful to make use of the factory and a shared validator instance, not sure if it matters in this case b.c. I think the splits made if folds is an int are deterministic, but if you had random splits in your cross val and build them separately for each model this might not be ideal.